### PR TITLE
core/socket: logs address or path which fails to be initialized

### LINKED ISSUE #11348: Log failed resource for socket units
**Is your feature request related to a problem? Please describe.**
A socket unit can failed with the following message:
```
janv. 07 11:34:48 belette64 systemd[1]: Failed to create listening socket: Address already in use
janv. 07 11:34:48 belette64 systemd[1]: rpcbind.socket: Failed to listen on sockets: Address already in use
janv. 07 11:34:48 belette64 systemd[1]: rpcbind.socket: Failed with result 'resources'.
```
However, in this case `rpcbind.socket` have 8 `ListenStream` and 7 `ListenDatagram` directives. This is very hard to tell which resource(s) failed.

**Describe the solution you'd like**
The failed resource should be logged in the journal. E.g.:
```
Failed to create listening socket: Address already in use (/run/rpcbind.sock)
```

**Describe alternatives you've considered**
Using `strace -e trace=%network -yy -p1`. This is not nice.

### PR DESCRIPTION
**Pull request by yuwata**
Closes #11348.

(Not tested.)

### CODE REVIEW

File: src/core/socket.c [hunk 1]
```diff
@@ -1593,59 +1596,58 @@ static int socket_open_fds(Socket *s) {
...
-                        p->fd = special_address_create(p->path, s->writable);
-                        if (p->fd < 0) {
-                                r = p->fd;
-                                goto rollback;
-                        }
+                        r = special_address_create(p->path, s->writable);
+                        if (r < 0)
+                                return r;
+
+                        p->fd = r;
```
Comment (keszybz) [inline src/core/socket.c:1614]: There's no need to go through `r` here:
```C
p->fd = special_address_create(p->path, s->writable);
if (p->fd < 0)
          return p->fd;
```

File: src/core/socket.c [hunk 2]
```diff
@@ -1593,59 +1596,58 @@ static int socket_open_fds(Socket *s) {
...
                                         s->directory_mode,
                                         s->socket_mode);
-                        if (p->fd < 0) {
-                                r = p->fd;
-                                goto rollback;
-                        }
+                        if (r < 0)
+                                return r;
 
+                        p->fd = r;
```
Comment (keszybz) [inline src/core/socket.c:1626]: Same here and below...

**Comment by poettering** [comment]
lgtm too

File: src/core/socket.c (hunk 1 above)
Comment (yuwata) [reply src/core/socket.c:1614]: OK. I've posted a follow-up commit for this in #11361.

**Pull request merged**